internal/driver/glfw/menu_darwin.go

```diff
@@ -100,3 +105,27 @@ func addNativeMenu(w *window, menu *fyne.Menu, nextItemID int, prepend bool) int
 	}
 	return nextItemID
 }
+
+func addNativeSubMenu(w *window, nsParentMenuItem unsafe.Pointer, menu *fyne.Menu, nextItemID int) int {
+	nsMenu := C.createDarwinMenu(C.CString(menu.Label))
+	for _, item := range menu.Items {
+		nsMenuItem := C.insertDarwinMenuItem(
+			nsMenu,
+			C.CString(item.Label),
+			C.int(nextItemID),
+			C.int(-1),
+			C.bool(item.IsSeparator),
+		)
+		if item.ChildMenu != nil {
+			nextItemID = addNativeSubMenu(w, nsMenuItem, item.ChildMenu, nextItemID)
+		}
+		if !item.IsSeparator {
+			if action := item.Action; action != nil { // catch action value
```

Comment:
this callback handling is a duplicate of above, maybe shouldn't be?

Reply:
I’ll DRY up …